docs/LibraryEvolution.rst

```diff
@@ -932,8 +912,8 @@ little more restrictive than structs; they only allow the following changes:
 
 Finally, classes allow the following changes that do not apply to structs:
 
-- A class may be marked ``open`` if it is not already marked ``final``.
-- A class may be marked ``final`` if it is not already marked ``open``.
+- A public class may be made ``open`` if it is not already marked ``final``.
+- A non-``open`` public class may be marked ``final``.
 - Removing an explicit deinitializer. (A class with no declared deinitializer
```

Comment:
Any reason adding an explicit deinitializer is not safe?

Reply:
Whoops, yes, that should be there too. I wonder how I missed that the first time.

(Or maybe I was worried about us using it to optimize, but you already can't do that if you can add members.)